managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -320,9 +319,44 @@ public ManagedLedgerImpl(ManagedLedgerFactoryImpl factory, BookKeeper bookKeeper
         }
     }
 
-    synchronized void initialize(final ManagedLedgerInitializeLedgerCallback callback, final Object ctx) {
+    synchronized void initialize(final ManagedLedgerInitializeLedgerCallback originalCb, final Object ctx) {
         log.info("Opening managed ledger {}", name);
 
+        ManagedLedgerInitializeLedgerCallback callback = new ManagedLedgerInitializeLedgerCallback() {
+            @Override
+            public void initializeComplete() {
+                if (lastConfirmedEntry != null && lastConfirmedEntry.entryId != -1) {
+                    // check that ledger used by lastConfirmedEntry actually exists
+                    try {
+                        getLedgerMetadata(lastConfirmedEntry.getLedgerId()).get();
```

Comment:
should we chain this CompletableFuture and call` originalCb.initializeComplete();` on completion ?
otherwise we are introducing a blocking call in the chain, also getLedgerMetadata will be a blocking call related to ZK client

```
getLedgerMetadata(lastConfirmedEntry.getLedgerId()).thenRun( () -> {
      originalCb.initializeComplete();
}).exceptionally((err) -> {
     originalCb.initializeFailed(err);
     ....
});
```

Reply:
fixed